src/main/java/com/vinyl/service/Validation.java

```diff
@@ -0,0 +1,23 @@
+package com.vinyl.service;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.vinyl.model.User;
+import com.vinyl.service.exception.BadRequestException;
+
+public class Validation {
```

Comment:
Validator

Reply:
make it a spring bean and inject into the service

Reply:
fixed